x-pack/legacy/plugins/ml/public/jobs/new_job_new/common/chart_loader/chart_loader.ts

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { SavedSearch } from 'src/legacy/core_plugins/kibana/public/discover/types';
+import { IndexPatternWithType, IndexPatternTitle } from '../../../../../common/types/kibana';
+import { Field, SplitField, AggFieldPair } from '../../../../../common/types/fields';
+import { ml } from '../../../../services/ml_api_service';
+import { mlResultsService } from '../../../../services/results_service';
+import { getCategoryFields } from './searches';
+
+type DetectorIndex = number;
+export interface LineChartPoint {
+  time: number | string;
+  value: number;
+}
+type SplitFieldValue = string | null;
+export type LineChartData = Record<DetectorIndex, LineChartPoint[]>;
+
+export class ChartLoader {
+  protected _indexPattern: IndexPatternWithType;
+  protected _savedSearch: SavedSearch;
+  protected _indexPatternTitle: IndexPatternTitle = '';
+  protected _timeFieldName: string = '';
+  protected _query: object = {};
+
+  constructor(indexPattern: IndexPatternWithType, savedSearch: SavedSearch, query: object) {
+    this._indexPattern = indexPattern;
+    this._savedSearch = savedSearch;
+    this._indexPatternTitle = indexPattern.title;
+    this._query = query;
+
+    if (typeof indexPattern.timeFieldName === 'string') {
+      this._timeFieldName = indexPattern.timeFieldName;
+    }
+  }
+
+  async loadLineCharts(
+    start: number,
+    end: number,
+    aggFieldPairs: AggFieldPair[],
+    splitField: SplitField,
+    splitFieldValue: SplitFieldValue,
+    intervalMs: number
+  ): Promise<LineChartData> {
+    if (this._timeFieldName !== '') {
+      const splitFieldName = splitField !== null ? splitField.name : null;
+
+      const resp = await ml.jobs.newJobLineChart(
+        this._indexPatternTitle,
+        this._timeFieldName,
+        start,
+        end,
+        intervalMs,
+        this._query,
+        aggFieldPairs.map(af => ({
+          agg: af.agg.dslName,
+          field: af.field.name,
+        })),
+        splitFieldName,
+        splitFieldValue
+      );
+      return resp.results;
+    }
+    return {};
+  }
+
+  async loadPopulationCharts(
+    start: number,
+    end: number,
+    aggFieldPairs: AggFieldPair[],
+    splitField: SplitField,
+    intervalMs: number
+  ): Promise<LineChartData> {
+    if (this._timeFieldName !== '') {
+      const splitFieldName = splitField !== null ? splitField.name : '';
+
+      const resp = await ml.jobs.newJobPopulationsChart(
+        this._indexPatternTitle,
+        this._timeFieldName,
+        start,
+        end,
+        intervalMs,
+        this._query,
+        aggFieldPairs.map(af => {
```

Comment:
Nit: would make it more readable to move this out into a var instead of doing the map inline.